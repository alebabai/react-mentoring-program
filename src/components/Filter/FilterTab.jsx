import React from 'react'
import classNames from 'classnames'

import styles from './style.css'

export const FilterTab = ({ id, title = id, active, onClick }) => (
    <div key={id} className={classNames(styles.tab, { [styles.tabActive]: active })} onClick={onClick}>{title}</div>
)

export default FilterTab
