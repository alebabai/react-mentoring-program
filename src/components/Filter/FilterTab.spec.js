import React from 'react'

import FilterTab from './FilterTab'

import renderer from 'react-test-renderer';

describe('Should match snapshot', () => {
    test('with all props', () => {
        const component = renderer.create(
            <FilterTab id="genres" title="Genres" active={true} onClick={() => { console.log('Tab clicked') }} />
        )
        expect(component.toJSON()).toMatchSnapshot()
    })
    test('with partial props', () => {
        const component = renderer.create(
            <FilterTab id="genres" onClick={() => { console.log('Tab clicked') }} />
        )
        expect(component.toJSON()).toMatchSnapshot()
    })
})
