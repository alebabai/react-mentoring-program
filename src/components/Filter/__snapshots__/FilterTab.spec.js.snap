// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot with all props 1`] = `
<div
  className="tab tabActive"
  onClick={[Function]}
>
  Genres
</div>
`;

exports[`Should match snapshot with partial props 1`] = `
<div
  className="tab"
  onClick={[Function]}
>
  genres
</div>
`;
